Accounts styles

jgoley opened the pull request:
I think we should consider asking for additional information about users ie location.
I suppose in the future we will list organizations, teams and data sets in a user's profile?

aslagle commented on packages/controllers/accounts_header_buttons.coffee:
```diff
@@ -1,3 +1,9 @@
+  @state = new ReactiveVar("signIn")
+
+Template.accountsHeaderButtons.helpers
+  state: -> Template.instance().state
```
bad merge

jgoley replied:
my bad

jgoley replied:
Oh wait...that's a different controller

jgoley replied:
Should i rename both instances?

aslagle replied:
I thought it wasn't supposed to be in this controller at all anymore.

jgoley replied:
True. I'll remove it.

aslagle commented on packages/views/profile_detail.jade:
```diff
@@ -1,10 +1,18 @@
-          if userProfile.emailAddress
-            p Email Address: #{userProfile.emailAddress}
+      .col-sm-4.space-btm-3
+        .avatar.space-btm-2(style="background-image: url('http://placehold.it/800x800')")
```
are we going to have avatars?

jgoley replied:
I thought we wanted profile images. maybe that is a bad name for it?

aslagle replied:
Yeah, it's fine - but I don't think we should have a placeholder, in case we don't get around to it.

jgoley commented:
Forgot to style the login/register modal. I'll do that before this merges.

jgoley commented:
I've styled the sign-in/register modal. Oh and also removed the placeholder image and duplicative helper.

jgoley commented:
I've resolved the conflicts causing a few of the tests to fail. Now it looks like there's an issue with connecting to the selenium server. @frosario is there a way to restart the Travis CI build to see if its resolved in another?

frosario commented:
Clicking on the Details link to the right of "Some Checks have failed" will take you the failing job.
Once there, you can click on circular button to have travis retry.
The button looks like: 
![button](https://cloud.githubusercontent.com/assets/513268/8677923/c9a92eba-2a1f-11e5-9828-7ca44a4c5a0b.png)

jgoley commented:
I'm not seeing it: 
![screen shot 2015-07-14 at 12 04 13 pm](https://cloud.githubusercontent.com/assets/4105343/8678092/9b3236a2-2a20-11e5-8f6c-060c716e5d58.png)

frosario commented:
You may need to sync your travis profile. Other uses have run into this issue for some reason. Go to: `https://travis-ci.org/profile/` and press the sync button. It may take a few minutes for the your travis account to sync with your github account. Not sure what causes this lag with travis.

frosario commented:
Now the tests pass, so I think selenium (or maybe phantomjs) is not starting up fast enough before the test run.

jgoley commented:
Yep, that was it. Thanks!

aslagle commented on packages/controllers/accounts_modal.coffee:
```diff
@@ -1,4 +1,5 @@
 AccountsTemplates.configure
+  showPlaceholders: false
```
What is this for?

jgoley replied:
I thought the placeholders inside the inputs were redundant—they were identical to the input's label.

aslagle commented on packages/styles/accounts.import.styl:
```diff
@@ -0,0 +1,55 @@
+  h3
+    padding 10px
+    margin 0
+    color white
```
If the title is hidden, why does it have styling on the h3?

jgoley replied:
Good catch. I waffled on whether we needed the title. Will remove

aslagle commented on packages/styles/accounts.import.styl:
```diff
@@ -0,0 +1,55 @@
+  text-align center
+
+.at-btn
+  makeButton($confirm)
```
Should these buttons be confirm-colored or something more neutral? I think the green works well when it's paired with red or yellow buttons, but I'm not sure about it by itself.

aslagle replied:
Also the register/sign up links are reddish and I'm not sure how that ties in to the color scheme.

jgoley replied:
I guess was thinking about it outside of the bounds of bootstrap—green is more appropriate for a call to action but it escapes the pattern bootstrap establishes. I'll use the primary color instead.

jgoley replied:
Yeah I set the active link color to the red of the color scheme. Which may not be the best choice, but I don't want to use either the primary or secondary. I can come up tertiary color for links and such that's less 'dangerous'.

aslagle commented on packages/styles/globals.import.styl:
```diff
@@ -85,6 +113,49 @@ textarea.form-control
+  display inline-block
+  vertical-align middle
+
+.toggleSwitch
```
Is this used for something?

jgoley replied:
Oh I thought I removed that. It probably will be in the future and I was preparing to use it for the email hiding option.

aslagle commented on packages/styles/header.import.styl:
```diff
@@ -37,6 +37,9 @@ header
   li
+    &.open
+        > a
+          background $primary !important
```
Is this used for something? Why !important?

aslagle commented on packages/styles/profile.import.styl:
```diff
@@ -0,0 +1,37 @@
+    font-size 130%
+    color $secondary
+  .avatar
+    min-width 100%
```
Avatar is still in here

aslagle commented on packages/styles/profile.import.styl:
```diff
@@ -0,0 +1,37 @@
+.avatar
+  min-height 250px
+  background-size cover
+  background-position center center
```
Avatar is here too

aslagle commented on packages/views/accounts_header_buttons.jade:
```diff
@@ -1,11 +1,18 @@
+    li.dropdown
+      a.dropdown-toggle.admin-settings(href='#' data-toggle='dropdown' aria-expanded='false')
+        span Account
+        i.fa.fa-gear
```
I'm not sure it's obvious enough what Account + gear is....

jgoley replied:
If it's a dropdown or the content of the dropdown?

aslagle replied:
Both I guess - for example people might not be used to sign out being in something like that.

jgoley replied:
I think nesting "sign out" in a dropdown is relatively common (Sign-in or register is more important which is why it has a dedicated button—I doubt most users log out of applications too often.) I'll replace the gear with a down arrow. I suppose we can change this in the future as the navigation builds?

aslagle replied:
What about putting their name or email instead of "Account"? That seems pretty common.

jgoley replied:
That's a good idea.

aslagle commented on packages/styles/variables.import.styl:
```diff
@@ -79,5 +79,5 @@ $min-font           = 12px
 $min-width          = 768px
 $max-width          = 1200px
-$vertical-rhythm    = true
+$vertical-rhythm    = false
```
What is vertical-rhythm? It sounds cool

jgoley replied:
http://webtypography.net/2.2.2
It sets a margin on headers which conflicts with other styles.

jgoley commented:
@aslagle I've updated things.
Good call with changing "Account" to the current user's name. (I followed Danny's pattern)
![screen shot 2015-07-14 at 3 35 20 pm](https://cloud.githubusercontent.com/assets/4105343/8682987/03efccfa-2a3e-11e5-86d1-60f447ef73ab.png)

I haven't changed the link color—I think for now its okay. It needs have sufficient contrast from the text and other colors.

jgoley commented:
Well I thought I had this correct but publishing the userProfile info in the header seems to make the email address always visible.
Here's what I currently have but `userInfo` is undefined
```
if Meteor.isClient
  Template.header.onCreated ->
    @subscribe('currentUserInfo', Meteor.userId())
    @accountsState = new ReactiveVar("signIn")

  Template.header.helpers
    accountsState: -> Template.instance().accountsState
    userInfo: -> UserProfiles.findOne({userId: Meteor.userId()})

if Meteor.isServer
  Meteor.publish 'currentUserInfo', (id) ->
    UserProfiles.findOne({userId: id}, {fields: {fullName: 1}})
```

aslagle commented:
You need to change package.js to have header.coffee on the server, it wasn't needed before.

jgoley commented:
Ah, I see. Thanks!

aslagle commented on packages/controllers/header.coffee:
```diff
@@ -1,5 +1,13 @@
-  @accountsState = new ReactiveVar("signIn")
+if Meteor.isClient
+  Template.header.onCreated ->
+    @subscribe('currentUserName', Meteor.userId())
```
It would make more sense to put the currentUserName stuff in accountsHeaderButtons if that's the only part that uses it.

jgoley replied:
Good idea. Just made this change.

aslagle commented on packages/styles/header.import.styl:
```diff
@@ -99,6 +110,62 @@ header
+        padding 8px 20px
+        text-shadow none
+        +below(1)
+          padding 8px 35px !important
```
what's this overriding?

jgoley replied:
I think I structured the stylus haphazardly when implementing rupture. I did a good bit of refactoring to hopefully clear things up. The styles are still pretty complex but its mostly because of bootstrap's nested selections.

aslagle commented on packages/styles/variables.import.styl:
```diff
@@ -28,7 +28,7 @@ $confirm              = rgba(103, 169, 104, 1) // Green
 $pending              = rgba(239, 172, 86, 1) // Yellow
 
-$header-footer-BG     = $secondary
+$header-footer-BG     = $secondary //lighten(desaturate($secondary, 50%), 10%)
```
this comment?

aslagle commented on packages/styles/globals.import.styl:
```diff
@@ -4,7 +4,7 @@ body
 body
   font-family inherit
-  > div
+  id['__flow-root']
```
What does this change fix?

jgoley replied:
I was having trouble positioning the toastr since its added to the body. It needs to be more specific. I suppose if it will always be body's first descendant we could use the first-child selector instead of the id.

aslagle replied:
Ah I see. It would be better not to depend on flow layout in here, so I guess I'd prefer the first child. But a different layout package might structure things differently and require changes anyway.

jgoley replied:
Is there a way to add a class to the wrapper div (`__flow-root`)?

aslagle replied:
Yeah, you could put in another div and set it to be the root node: https://github.com/meteorhacks/flow-layout#set-different-root-node

jgoley commented:
I changed the root node to the body and added a page-wrapper so its similar to previous projects. Toasts appear on the right under the header and above the header on small screens.

aslagle commented:
I'm seeing 'Account' instead of the username on the organizations and the splash page - I think this is a bug.

jgoley commented:
I see...it doesn't seem to work after the user signs in but does after subsequent sessions.

aslagle commented:
Oh it's because of this:
```
Template.accountsHeaderButtons.onCreated ->
    @subscribe('currentUserName', Meteor.userId())
```
The subscribe needs to be in an autorun so it updates when the userId changes.

aslagle commented:
This looks good to me.
